sdc/datatypes/hpat_pandas_dataframe_functions.py

```diff
@@ -1279,8 +1279,7 @@ def df_getitem_slice_idx_main_codelines(self, idx):
     for i, col in enumerate(self.columns):
         res_data = f'res_data_{i}'
         func_lines += [
-            f'  data_{i} = get_dataframe_data(self, {i})',
-            f'  {res_data} = pandas.Series(data_{i}[idx], index=res_index[idx], name="{col}")'
+            f'  {res_data} = pandas.Series((self._data[{i}])[idx], index=res_index[idx], name="{col}")'
```

Comment:
Maybe skip constructing of Series and use array directly?

Reply:
I've tried and it causes some tests to fail. And also it reduces ir only for about 10Mb (175->165Mb)